http/client_lens.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package http
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/sourcenetwork/immutable/enumerable"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/datastore"
+)
+
+var _ client.LensRegistry = (*LensRegistry)(nil)
+
+// LensRegistry implements the client.LensRegistry interface over HTTP.
+type LensRegistry struct {
+	http *httpClient
+}
+
+func (c *LensRegistry) WithTxn(tx datastore.Txn) client.LensRegistry {
+	http := c.http.withTxn(tx.ID())
+	return &LensRegistry{http}
+}
+
+func (c *LensRegistry) SetMigration(ctx context.Context, config client.LensConfig) error {
+	methodURL := c.http.baseURL.JoinPath("lens")
+
+	body, err := json.Marshal(config)
+	if err != nil {
+		return err
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, methodURL.String(), bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+	_, err = c.http.request(req)
+	return err
+}
+
+func (c *LensRegistry) ReloadLenses(ctx context.Context) error {
+	methodURL := c.http.baseURL.JoinPath("lens", "reload")
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, methodURL.String(), nil)
+	if err != nil {
+		return err
+	}
+	_, err = c.http.request(req)
+	return err
+}
+
+func (c *LensRegistry) MigrateUp(
+	ctx context.Context,
+	src enumerable.Enumerable[map[string]any],
+	schemaVersionID string,
+) (enumerable.Enumerable[map[string]any], error) {
+	methodURL := c.http.baseURL.JoinPath("lens", schemaVersionID, "up")
+
+	body, err := json.Marshal(src)
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, methodURL.String(), bytes.NewBuffer(body))
+	if err != nil {
+		return nil, err
+	}
+	var result enumerable.Enumerable[map[string]any]
+	if err := c.http.requestJson(req, &result); err != nil {
```

Comment:
question: Are you 100% sure this works? You appear to be unmarshalling json to an interface.  If it does work, how, and what is the resultant concrete type?  If it does not work, did the tests miss this, or did you forget to run them?  If they missed it, how?

`MigrateUp`, `MigrateDown` on `LensRegistry` and `lensHandler` appear to do this.

Reply:
I think I know why this was missed.

Previously we have been able to rely on the fact that we only ever had one implementation of the client interfaces, this allowed us to get away with not directly testing most functions implemented, so long as they were called internally  by functions that were directly tested.

These functions (as do most of them, our testing is horrible across the go-embedded stuff) fall into that category. They have no tests targeting them. 

I think this requires discussion at standup, as it means that significant chunks of this PR are untested, as will likely much of the CLI stuff be. There is a significant risk of other bugs besides the probable one here that I have flagged.